atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/metrics/SweepMetricsManagerTest.java

```diff
@@ -110,204 +107,131 @@ public void tearDown() {
+        sweepMetricsManager.updateMetrics(SWEEP_RESULTS, TABLE_REF);
 
-        assertRecordedHistogramTaggedUnSafeOneIteration(CELLS_EXAMINED, EXAMINED);
-        assertRecordedHistogramTaggedUnSafeOneIteration(CELLS_SWEPT, DELETED);
-        assertRecordedHistogramTaggedUnSafeOneIteration(TIME_SPENT_SWEEPING, TIME_SWEEPING);
-    }
+        assertRecordedCurrentValue(ImmutableMap.of(
+                CELLS_EXAMINED, EXAMINED,
+                CELLS_SWEPT, DELETED,
+                TIME_SPENT_SWEEPING, TIME_SWEEPING,
+                TABLE_BEING_SWEPT, UNSAFE_FULLY_QUALIFIED));
 
-    @Test
-    public void allMetersAreRecordedForOneIteration() {
-        sweepMetricsManager.updateMetrics(SWEEP_RESULTS, TABLE_REF, UpdateEventType.ONE_ITERATION);
+        sweepMetricsManager.updateMetrics(OTHER_SWEEP_RESULTS, TABLE_REF2);
 
-        assertRecordedMeterNonTaggedOneIteration(CELLS_EXAMINED, EXAMINED);
-        assertRecordedMeterNonTaggedOneIteration(CELLS_SWEPT, DELETED);
-        assertRecordedMeterNonTaggedOneIteration(TIME_SPENT_SWEEPING, TIME_SWEEPING);
+        assertRecordedCurrentValue(ImmutableMap.of(
+                CELLS_EXAMINED, OTHER_EXAMINED,
+                CELLS_SWEPT, OTHER_DELETED,
+                TIME_SPENT_SWEEPING, OTHER_TIME_SWEEPING,
+                TABLE_BEING_SWEPT, UNSAFE_FULLY_QUALIFIED));
     }
 
     @Test
-    public void timeElapsedCurrentValueIsRecordedForOneIteration() {
-        sweepMetricsManager.updateMetrics(SWEEP_RESULTS, TABLE_REF, UpdateEventType.ONE_ITERATION);
+    public void allGaugesAreSetForUnknownSafetyAsUnsafe() {
+        setLoggingSafety(ImmutableMap.of());
+        sweepMetricsManager.updateMetrics(SWEEP_RESULTS, TABLE_REF);
 
-        assertSweepTimeElapsedCurrentValueWithinMarginOfError(START_TIME);
-    }
+        assertRecordedCurrentValue(ImmutableMap.of(
+                CELLS_EXAMINED, EXAMINED,
+                CELLS_SWEPT, DELETED,
+                TIME_SPENT_SWEEPING, TIME_SWEEPING,
+                TABLE_BEING_SWEPT, UNSAFE_FULLY_QUALIFIED));
 
-    @Test
-    public void allNonTaggedHistogramsAreRecordedForFulTable() {
-        sweepMetricsManager.updateMetrics(SWEEP_RESULTS, TABLE_REF, UpdateEventType.FULL_TABLE);
+        sweepMetricsManager.updateMetrics(OTHER_SWEEP_RESULTS, TABLE_REF2);
 
-        assertRecordedHistogramNonTaggedFullTable(CELLS_EXAMINED, TABLE_REF, EXAMINED);
-        assertRecordedHistogramNonTaggedFullTable(CELLS_SWEPT, TABLE_REF, DELETED);
-        assertRecordedHistogramNonTaggedFullTable(TIME_SPENT_SWEEPING, TABLE_REF, TIME_SWEEPING);
-        assertSweepTimeElapsedHistogramWithinMarginOfError(START_TIME);
+        assertRecordedCurrentValue(ImmutableMap.of(
+                CELLS_EXAMINED, OTHER_EXAMINED,
+                CELLS_SWEPT, OTHER_DELETED,
+                TIME_SPENT_SWEEPING, OTHER_TIME_SWEEPING,
+                TABLE_BEING_SWEPT, UNSAFE_FULLY_QUALIFIED));
```

Comment:
nit: some of these blocks feel like they can be extracted to a common helper, though as this is test code I don't actually mind the verbosity too much.